packer/builder/azure/arm/step_create_resource_group.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources"
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+type StepCreateResourceGroup struct {
+	client *AzureClient
+	create func(resourceGroupName string, location string) error
+	say    func(message string)
+	error  func(e error)
+}
+
+func NewStepCreateResourceGroup(client *AzureClient, ui packer.Ui) *StepCreateResourceGroup {
+	var step = &StepCreateResourceGroup{
+		client: client,
+		say:    func(message string) { ui.Say(message) },
+		error:  func(e error) { ui.Error(e.Error()) },
+	}
+
+	step.create = step.createResourceGroup
+	return step
+}
+
+func (s *StepCreateResourceGroup) createResourceGroup(resourceGroupName string, location string) error {
+	_, err := s.client.GroupsClient.CreateOrUpdate(resourceGroupName, resources.ResourceGroup{
+		Location: &location,
+	})
+
+	return err
+}
+
+func (s *StepCreateResourceGroup) Run(state multistep.StateBag) multistep.StepAction {
+	s.say("Creating resource group ...")
+
+	var resourceGroupName = state.Get(constants.ArmResourceGroupName).(string)
+	var location = state.Get(constants.ArmLocation).(string)
+
+	s.say(fmt.Sprintf(" -> ResourceGroupName : '%s'", resourceGroupName))
+	s.say(fmt.Sprintf(" -> Location          : '%s'", location))
+
+	err := s.create(resourceGroupName, location)
+	if err != nil {
+		state.Put(constants.Error, err)
+		s.error(err)
+
+		return multistep.ActionHalt
+	}
+
+	return multistep.ActionContinue
+}
+
+func (*StepCreateResourceGroup) Cleanup(multistep.StateBag) {
```

Comment:
You probably want to clean up the resource group in here. (This is also only half-used in the SMAPI version.)